client.go

```diff
@@ -1830,3 +1836,33 @@ func flags(f int) uint32 {
 	}
 	return out
 }
+
+// posixPerm converts Go permission bits to POSIX permission bits,
+// masking off file type bits.
+//
+// This differs from fromFileMode in that it preserves the POSIX versions of
+// setuid, setgid and sticky in m, because we've historically supported those
+// bits.
+func posixPerm(m os.FileMode) uint32 {
+	// POSIX mode bits, as listed at
+	// https://tools.ietf.org/html/draft-ietf-secsh-filexfer-13#section-7.6
+	const (
+		posixSetuid = 04000
```

Comment:
This code should probably instead go into `stat_posix.go` and `stat_plan9.go` so that we can use the appropriate `syscall` values the same as `fromFileMode` and handle plan9 appropriately.

Reply:
The problem with that approach is that we get the bits from the platform the client is running on, rather than the bits that the server expects.

Reply:
I have checked all these values. Except for `S_IFMT` these values are all the same across all implementations. (They have been after all codifed by POSIX after all.)

Reply:
fromFileMode on Plan 9 also doesn't handle the bits 07000 at all, nor does syscall on Plan 9 have the S_ISUID etc. constants. For the server, that's ok since Plan 9 doesn't have setuid, but it means a Plan 9 client talking to a Unix server couldn't set these bits.

Adding them in P9 fromFileMode runs the risk of returning some erroneous bits from Plan 9 servers, if it happens to use 07000 for a different purpose...

Reply:
The benefits of putting them into their specific files, is that the `stat_posix.go` file can use the definied `syscall` values, while the `stat_plan9.go` file can define these otherwise-POSIX values individually.

This means the code is _intentional_ rather than assuming that plan9 is unlimitedly the same.

Reply:
The client should do exactly the same on all platforms, though. The syscall package is for talking to your local kernel. This code is talking to a remote.

Reply:
To put it differently, I'm regarding the value to be sent as part of the *protocol*, and the protocol is conceptually different from the syscall interface. I'll admit the end result is practically the same, except that it also prevents having to write plan9-specific code that I cannot test.